client/src/templates/Challenges/components/completion-modal.tsx

```diff
@@ -225,135 +208,79 @@ export class CompletionModalInner extends Component<
+            />
+          </Modal.Body>
+          <Modal.Footer>
+            {isSignedIn ? null : (
+              <Login block={true}>{t('learn.sign-in-save')}</Login>
+            )}
             <Button
               block={true}
-              bsSize='lg'
+              bsSize='large'
               bsStyle='primary'
-              className='btn-invert'
-              download={`${dashedName}.txt`}
-              href={this.state.downloadURL}
+              onClick={() => this.handleSubmit()}
             >
-              {t('learn.download-solution')}
+              {isSignedIn
+                ? t('buttons.submit-and-go')
+                : t('buttons.go-to-next')}
+              <span className='hidden-xs'> (Ctrl + Enter)</span>
             </Button>
-          ) : null}
-        </Modal.Footer>
-      </Modal>
+            {this.state.downloadURL ? (
+              <Button
+                block={true}
+                bsSize='lg'
+                bsStyle='primary'
+                className='btn-invert'
+                download={`${dashedName}.txt`}
+                href={this.state.downloadURL}
+              >
+                {t('learn.download-solution')}
+              </Button>
+            ) : null}
+          </Modal.Footer>
+        </Modal>
+      </div>
     );
   }
 }
 
-interface Options {
-  isFinalProjectBlock: boolean;
-}
-
-interface CertificateNode {
-  challenge: {
-    // TODO: use enum
-    certification: string;
-    tests: { id: string }[];
-  };
-}
-
-const useCurrentBlockIds = (
-  block: string,
-  certification: string,
-  options?: Options
-) => {
-  const {
-    allChallengeNode: { edges: challengeEdges },
-    allCertificateNode: { nodes: certificateNodes }
-  }: {
-    allChallengeNode: AllChallengeNode;
-    allCertificateNode: { nodes: CertificateNode[] };
-  } = useStaticQuery(graphql`
-    query getCurrentBlockNodes {
-      allChallengeNode(
-        sort: {
-          fields: [
-            challenge___superOrder
-            challenge___order
-            challenge___challengeOrder
-          ]
-        }
-      ) {
-        edges {
-          node {
-            challenge {
-              block
-              id
-            }
-          }
-        }
-      }
-      allCertificateNode {
-        nodes {
-          challenge {
-            certification
-            tests {
-              id
-            }
-          }
-        }
-      }
-    }
-  `);
-
-  const currentCertificateIds = certificateNodes
-    .filter(
-      node => dasherize(node.challenge.certification) === certification
-    )[0]
-    ?.challenge.tests.map(test => test.id);
-  const currentBlockIds = challengeEdges
-    .filter(edge => edge.node.challenge.block === block)
-    .map(edge => edge.node.challenge.id);
-
-  return options?.isFinalProjectBlock ? currentCertificateIds : currentBlockIds;
-};
-
 const CompletionModal = (props: CompletionModalsProps) => {
   const currentBlockIds = useCurrentBlockIds(
```

Comment:
> Is it not the case that the completedPercent is calculated in the show component ?
blockIds are being queried here to set the state of parent component. That should not be necessary. 

The BlockIds are queried here and set as state on the parent component using setCurrentBlockIds.

If BlockIds are needed on the parent component, query them there and pass only what is necessary to children components.